autogpt_platform/backend/backend/util/settings.py

```diff
@@ -115,6 +115,30 @@ class Config(UpdateTrackingModel["Config"], BaseSettings):
         description="The port for agent server API to run on",
     )
 
+    backend_cors_allow_origins: List[str] = Field(default_factory=list)
+
+    @field_validator("backend_cors_allow_origins")
+    @classmethod
+    def validate_cors_allow_origins(cls, v: List[str]) -> List[str]:
+        out = []
+        for url in v:
+            url = url.strip()
+            if url.startswith(("http://", "https://")):
+                out.append(url)
+            else:
+                raise ValueError(f"Invalid URL: {url}")
+
+        # If using local callback add both localhost and 127.0.0.1
+        # NOTE: Localhost does not use ssl
+        has_localhost = any(["localhost" in url for url in out])
+        has_127_0_0_1 = any(["127.0.0.1" in url for url in out])
+        if has_127_0_0_1 and not has_localhost:
```

Comment:
Why do we assuming port 3000?
Should this use env var too?

Reply:
I've updated this to extract the port from the localhost / 127.0.0.1 entry that triggered this rule